src/sentry/static/sentry/app/views/acceptOrganizationInvite.tsx

```diff
@@ -0,0 +1,220 @@
+import React, {MouseEvent} from 'react';
+import styled from 'react-emotion';
+import {browserHistory} from 'react-router';
+
+import {logout} from 'app/actionCreators/account';
+import {t, tct} from 'app/locale';
+import {urlEncode} from '@sentry/utils';
+import Alert from 'app/components/alert';
+import AsyncView, {AsyncViewProps, AsyncViewState} from 'app/views/asyncView';
+import Button from 'app/components/button';
+import ConfigStore from 'app/stores/configStore';
+import Link from 'app/components/links/link';
+import NarrowLayout from 'app/components/narrowLayout';
+import SettingsPageHeader from 'app/views/settings/components/settingsPageHeader';
+import space from 'app/styles/space';
+
+type InviteDetails = {
+  orgSlug: string;
+  needsAuthentication: boolean;
+  needs2fa: boolean;
+  needsSso: boolean;
+  existingMember: boolean;
+  ssoProvider?: string;
+};
+
+type Props = AsyncViewProps;
+type State = AsyncViewState & {
+  inviteDetails: InviteDetails;
+  accepting: boolean | undefined;
+  acceptError: boolean | undefined;
+};
+
+class AcceptOrganizationInvite<P extends Props, S extends State> extends AsyncView<P, S> {
+  getEndpoints(): [string, string][] {
+    const {memberId, token} = this.props.params;
+    return [['inviteDetails', `/accept-invite/${memberId}/${token}/`]];
+  }
+
+  getTitle() {
+    return t('Accept Organization Invite');
+  }
+
+  makeNextUrl(path: string) {
+    return `${path}?${urlEncode({next: window.location.pathname})}`;
+  }
+
+  handleLogout = async (e: MouseEvent) => {
+    e.preventDefault();
+    await logout(this.api);
+    window.location.replace(this.makeNextUrl('/auth/login/'));
+  };
+
+  handleAcceptInvite = async () => {
+    const {memberId, token} = this.props.params;
+
+    this.setState({accepting: true});
+    try {
+      await this.api.requestPromise(`/accept-invite/${memberId}/${token}/`, {
+        method: 'POST',
+      });
+      browserHistory.push(`/${this.state.inviteDetails.orgSlug}/`);
```

Comment:
* Should this leave a history entry?
* Can we change the URL since this will add a redirect - we may want like a default URL constant that points to issues stream and use `replaceRouterParams`

Reply:
I'm OK with the redirect for the tradeoff of not having to come back here if we want to change what the default home page is.